aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -197,6 +220,16 @@ void cpu_flash_attention(
   int64_t lStrideB = logsumexp.stride(0);
   int64_t lStrideM = logsumexp.stride(1);
   int64_t lStrideH = logsumexp.stride(2);
+  int64_t mStrideB =
+      (has_attn_mask && attn_mask.value().size(0) > 1)
+      ? attn_mask.value().stride(0)
+      : 0;
+  int64_t mStrideH =
+      (has_attn_mask && attn_mask.value().size(1) > 1)
+      ? attn_mask.value().stride(1)
+      : 0;
+  int64_t mStrideM =
+      has_attn_mask ? attn_mask.value().stride(2) : 0;
```

Comment:
Does the code work for both 3D and 4D `attn_mask`?

Reply:
Yes, the code work for 2d, 3d and 4d `attn_mask`. It would be firstly reshaped in `reshape_attn_mask_to_4d`.